src/tqec/sketchup/dae_parse.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import annotations
+
+import xml.etree.ElementTree as ET
+from dataclasses import dataclass
+import pathlib
+
+import numpy as np
+
+from tqec.exceptions import TQECException
+
+
+# All library nodes defined in the template file
+LIBRARY_NODE_TYPES = {
+    "xoz",
+    "zox",
+    "ozxh",
+    "oxzh",
+    "zoxh",
+    "ozx",
+    "xzx",
+    "zxoh",
+    "zxz",
+    "oxz",
+    "xzoh",
+    "xozh",
+    "zzx",
+    "xxz",
+    "zxx",
+    "xzo",
+    "xzz",
+    "zxo",
+}
+
+
+@dataclass(frozen=True)
+class NodePosition:
+    """Position of a node in 3D space.
+
+    The (x,y,z) axis correspond to the (R,G,B) axis in the SketchUp model.
+    """
+
+    x: int
+    y: int
+    z: int
+
+    def __str__(self):
+        return f"({self.x}, {self.y}, {self.z})"
+
+    def to_tuple(self):
+        return (self.x, self.y, self.z)
+
+
+@dataclass(frozen=True)
+class Node:
+    """A node in the SketchUp model with a specific type and position."""
+
+    node_type: str
+    position: NodePosition
+
+    def __post_init__(self):
+        if self.node_type not in LIBRARY_NODE_TYPES:
+            raise TQECException(f"Invalid node type: {self.node_type}.")
+
+    def __str__(self) -> str:
+        return f"{self.node_type} at {self.position}"
+
+
+def parse_dae_file_to_nodes(filepath: str | pathlib.Path) -> list[Node]:
+    """Parse a SketchUp exported DAE file and return a list of nodes.
+
+    Args:
+        filepath: Path to the DAE file.
+
+    Returns:
+        A list of Node objects representing the nodes in the SketchUp model.
+    """
+    nodes = []
+    tree = ET.parse(filepath)
+
+    _remove_namespace(tree)
```

Comment:
It seems like there is no namespace in the provided test files, is that line (and function) really needed?

Reply:
There is the default namespace defined in `<COLLADA xmlns="http://www.collada.org/2005/11/COLLADASchema" version="1.4.1">`. And according to the [documentation of xml](https://docs.python.org/3/library/xml.etree.elementtree.html):
 
> Also, if there is a [default namespace](https://www.w3.org/TR/xml-names/#defaulting), that full URI gets prepended to all of the non-prefixed tags.

We need to remove it here.

Reply:
Then I do not understand the implementation:
```py
def _remove_namespace(tree: ET.ElementTree):
    for elem in tree.iter():
        if "}" in elem.tag:
            elem.tag = elem.tag.split("}", 1)[1]
```

Why is the function searching for `}`?

Reply:
Default namespace will also be prepended to each element's tag in the form of `{url}` when parsing with `xml.etree.ElementTree`.

Reply:
Ok, got it, I am fine with it then :) Thanks for the explanation.